src/vmm/src/vmm_config/machine_config.rs

```diff
@@ -206,33 +159,63 @@ where
     Ok(val)
 }
 
-/// Deserialization function for the `cpu_template` field in `VmConfig` and `VmUpdateConfig`.
-/// This is called only when `cpu_template` is present in the JSON configuration.
-fn deserialize_cpu_template<'de, D, T>(_d: D) -> std::result::Result<T, D::Error>
-where
-    D: de::Deserializer<'de>,
-    T: Deserialize<'de>,
-{
-    // If this function was called it means that `cpu_template` was specified in
-    // the JSON. Return an error since `cpu_template` is not supported on aarch64.
-    #[cfg(target_arch = "aarch64")]
-    return Err(de::Error::invalid_value(
-        de::Unexpected::Enum,
-        &"CPU templates are not supported on aarch64",
-    ));
-
-    #[cfg(target_arch = "x86_64")]
-    T::deserialize(_d)
-}
-
+/// Enum that represents types of cpu templates available.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum CpuTemplateType {
+    /// Custom cpu template
+    Custom(CustomCpuTemplate),
+    /// Static cpu template
+    Static(StaticCpuTemplate),
 }
 
+/// Strongly typed structure that represents the configuration of the
+/// microvm that is used internally.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct VmConfig {
+    /// Number of vcpu to start.
+    pub vcpu_count: u8,
+    /// The memory size in MiB.
+    pub mem_size_mib: usize,
+    /// Enables or disabled SMT.
+    pub smt: bool,
+    /// A CPU template that it is used to filter the CPU features exposed to the guest.
+    pub cpu_template: Option<CpuTemplateType>,
```

Comment:
As far as my understanding this change wants to do the API changes towards user configurability for templates.

However, the best way to go further is to merge the v1n1 implementation independently without doing any change to VmConfig.

You can just add the V1N1 option to the already existent `CpuFeaturesTemplate` and look at how we introduce the `CustomTemplate` option to the user independently (in another follow up PR).